project/Controller.gd

```diff
@@ -0,0 +1,32 @@
+extends Node
+class_name Controller
+
+enum {JOYPAD, MOUSE}
+var type: int
+var dev_id: int
```

Comment:
These are the vars that will need to be set on each controller object that gets instantiated in multiplayer.